batch/test/failure_injecting_batch_client.py

```diff
@@ -0,0 +1,36 @@
+import aiohttp
+
+
+class FailureInjectingBatchClient:
```

Comment:
So the design of this is making the main code unnecessarily complicated.  The retry should be inside the request operations.  If you want to inject failures, you need to inject it into the session.